closed/src/java.base/share/classes/com/sun/crypto/provider/NativeChaCha20Cipher.java

```diff
@@ -962,7 +1000,23 @@ public synchronized int doUpdate(byte[] in, int inOff, int inLen, byte[] out,
 
                 if (in != null) {
                     Objects.checkFromIndexSize(inOff, inLen, in.length);
-                    int ret = nativeCrypto.ChaCha20Update(context, in, inOff, inLen, out, outOff, /*aadArray*/ null, /*aadArray.length*/ 0);
+
+                    // Check if there is any overlap with the input and output
+                    // buffers. If so, create a temporary copy of the input such
+                    // that it does not overlap while performing an update computation
+                    // within the openssl library.
+                    int ret = -1;
+                    if (ChaChaEngine.arraysOverlap(in, inOff, inOff + inLen, out, outOff, outOff + getOutputSize(inLen, false))) {
+                        byte[] newInArray = new byte[inLen];
+                        System.arraycopy(in, inOff, newInArray, 0, inLen);
```

Comment:
Suggest using `Arrays.copyOfRange()` (here and on line 1090):
```java
                        byte[] newInArray = Arrays.copyOfRange(in, inOff, inOff + inLen);
```

Reply:
Updated to use the method suggested instead of `System.arraycopy`.